crates/core/src/node/inner/node_executor.rs

```diff
@@ -532,182 +512,139 @@ impl NodeExecutorHandle {
-            Ok(result) => Ok(result),
-            Err(_) => anyhow::bail!("failed to remove block interval as node executor is dropped"),
-        }
     }
 
     /// Request [`NodeExecutor`] to enforce next block's base fee per gas. Waits for the change to take
     /// place. Block might still not be produced by then.
-    pub async fn enforce_next_base_fee_per_gas_sync(&self, base_fee: U256) -> anyhow::Result<()> {
-        let (response_sender, response_receiver) = oneshot::channel();
-        self.command_sender
-            .send(Command::EnforceNextBaseFeePerGas(base_fee, response_sender))
+    pub async fn enforce_next_base_fee_per_gas_sync(
+        &self,
+        base_fee: U256,
+    ) -> Result<(), AnvilNodeError> {
+        self.execute_with_response(|sender| Command::EnforceNextBaseFeePerGas(base_fee, sender))
             .await
-            .map_err(|_| {
-                anyhow::anyhow!(
-                    "failed to enforce next base fee per gas as node executor is dropped"
-                )
-            })?;
+    }
+
+    async fn execute_without_response(&self, command: Command) -> Result<(), AnvilNodeError> {
+        let action_name = command.readable_description();
+        self.command_sender.send(command).await.map_err(|_| {
+            let error_msg = error_msg_node_executor_dropped(&format!("request to {action_name}"));
+            anvil_zksync::node::generic_error!("{error_msg}")
+        })
+    }
+    async fn execute_with_response<R>(
+        &self,
+        command_gen: impl FnOnce(oneshot::Sender<R>) -> Command,
+    ) -> Result<R, AnvilNodeError> {
+        let (response_sender, response_receiver) = oneshot::channel();
+
+        let command = command_gen(response_sender);
+        let action_name = command.readable_description();
+        self.command_sender.send(command).await.map_err(|_| {
+            let error_msg = error_msg_node_executor_dropped(&format!("request to {action_name}"));
+            anvil_zksync::node::generic_error!("{error_msg}")
+        })?;
 
         match response_receiver.await {
             Ok(result) => Ok(result),
             Err(_) => {
-                anyhow::bail!("failed to enforce next base fee per gas as node executor is dropped")
+                let error_msg = error_msg_node_executor_dropped(&format!(
+                    "receive a response to the request to {action_name}"
+                ));
+                Err(anvil_zksync::node::generic_error!("{error_msg}"))
             }
         }
     }
 }
 
+/// Helper function to produce uniform error messages when node executor is
+/// dropped. This happens across several functions in this file.
+fn error_msg_node_executor_dropped(action: &str) -> String {
+    format!(
+        "Failed to {action} because node executor is dropped. \
+         Another error was likely propagated from the main execution loop. \
+         If this is not the case, please, report this as a bug."
+    )
+}
```

Comment:
Also JFYI Rust coding conventions suggest 100 characters as the line limit so no need to be so frugal with 80 :)